### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by i25959341**
Great change !

### CODE REVIEW

**Comment by muzykantov** [comment]
Thanks. :) I would like to refactor your code for my tasks, but it's not ready yet. I want to save compatability, but will add it again later when finish main changes, ok? Can I add MIT Licence to code?

**Comment by i25959341** [comment]
@muzykantov yes please on the MIT licence

**Comment by muzykantov** [comment]
Is it a good idea to merge my changes into your repository? I broke compatibility, but improve speed and memory usage (about 80% code rewritten). Made 100% test coverage and benchmarks. Also created README, LICENSE and AUTHORS files. 

What do you think about this? (Sorry for my English)

**Comment by i25959341** [comment]
The change looks amazing. Thanks so much!!

On the compatibility, maybe we can keep a changelog, I wrote an issue and can add one in the weekend.

https://keepachangelog.com/en/1.0.0/

File: errors.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package orderbook
+
```
Comment (i25959341) [inline errors.go:2]: Good stuff!

**Pull request merged**